tests/cupy_tests/sparse_tests/test_linalg.py

```diff
@@ -0,0 +1,41 @@
+import unittest
+
+import numpy
+import scipy.sparse.linalg  # NOQA
+
+import cupy.sparse.linalg  # NOQA
+from cupy import testing
+
+
+@testing.parameterize(*testing.product({
+    'dtype': [numpy.float32, numpy.float64],
+}))
+@testing.with_requires('scipy')
+class TestLsqr(unittest.TestCase):
+
+    def setUp(self):
+        self.A = numpy.random.randint(2, size=(10, 10))
+        self.b = numpy.random.randint(2, size=10)
+
+    @testing.numpy_cupy_allclose(atol=1e-3, sp_name='sp')
+    def test_csrmatrix(self, xp, sp):
+        A = xp.array(self.A).astype(self.dtype)
+        b = xp.array(self.b).astype(self.dtype)
+        A = A + A.T
```

Comment:
Is this function only supports symmetric ones?

Reply:
This function also supports asymmetric matrices. It seems symmetrics matrices are easy to pass the tests without a fault. That's why I inserted this line. Instead of using symmetric matrices, I added `retry` decorator.